threat_intel/util/http.py

```diff
@@ -273,10 +274,31 @@ def _wait_for_response(self, requests, to_json):
             A list of dicts if to_json, a list of grequest.response otherwise
         """
         all_responses = []
-
-        for request, response in zip(requests, grequests.map(requests)):
+        
+        retry = 0
+        max_retry = 10
+        responses = None
+        while retry <= max_retry and responses is None:
+            try:
+               retry = retry + 1
+               responses = grequests.map(requests)
+                # test our responses
+                for response in responses:
+                   # for some odd conditions, requests will neither return an object nor raise an error
+                   if not response:
+                       #raise an error so we trigger a retry
+                       responses = None
```

Comment:
I don't think you need this here, response is already None by line 280, it only changes the status if we get something from grequests.